src/protocol/requests.rs

```diff
@@ -138,7 +139,30 @@ pub struct CompletionParams {
 #[derive(Serialize, Deserialize, Clone)]
 pub struct CompletionContext {
     #[serde(rename = "triggerKind")]
-    trigger_kind: i32,
+    pub trigger_kind: i32,
     #[serde(rename = "triggerCharacter")]
-    trigger_character: Option<String>,
+    pub trigger_character: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct SignatureHelpParams {
+    #[serde(rename = "textDocument")]
+    pub text_document: TextDocumentIdentifier,
+    pub position: Position,
+    pub context: Option<SignatureHelpContext>,
+}
+
+#[derive(Serialize_repr, Deserialize_repr, Clone)]
+#[repr(u32)]
+pub enum SignatureHelpTriggerKind {
+    Invoked = 1,
+    TriggerCharacter = 2,
+    ContentChange = 3,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct SignatureHelpContext {
+    pub is_retrigger: bool,
+    pub trigger_character: Option<String>,
```

Comment:
should this be a char? when serialized, it will be a string in json

Reply:
There is no difference in JSON between a char and a string.

Reply:
cool, so char should be better here